pkg/controllers/metrics/pod/controller.go

```diff
@@ -112,9 +129,28 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 }
 
 func (c *Controller) record(ctx context.Context, pod *v1.Pod) {
+	// Record pods state metric
 	labels := c.labels(ctx, pod)
 	podGaugeVec.With(labels).Set(float64(1))
 	c.labelsMap.Store(client.ObjectKeyFromObject(pod), labels)
+
+	// Record pods startup time metric
+	getReadyCondition := func() *v1.PodCondition {
+		for _, condition := range pod.Status.Conditions {
+			if condition.Type == v1.PodReady {
+				return &condition
+			}
+		}
+		return nil
+	}
```

Comment:
I'd either separate this out to another helper function or just do the for loop like normal instead of creating a function to call it.